src/Engine/GameInstance.java

```diff
@@ -39,21 +39,24 @@
   private Weathers defaultWeather;
 
   private GameScenario gameScenario;
+  boolean isSecurityEnabled;
```

Comment:
I feel like this and probably a few other things (fog, default weather, etc?) could move to `GameRules` now that that exists.
This has the benefit of not needing to duplicate the fog value for `MapWindow`, which means we can easily update it during the course of the game.

That said, I think I'm fine leaving this as is for now?

Reply:
I mean... maybe. I feel like I considered that, but "need for passwords" sounds more like an attribute of the game, rather than a rule of engagement. It could go either place though, I'm sure. Maybe it will make more sense to move it once `GameRules` (or even `GameScenario`) are a little more mature.

Reply:
Agreed, honestly.
I think fog and weather should move, but not in this PR.